dist_tables/dml.rst

```diff
@@ -38,6 +38,105 @@ For example:
 
     If COPY fails to open a connection for a shard placement then it behaves in the same way as INSERT, namely to mark the placement(s) as inactive unless there are no more active placements. If any other failure occurs after connecting, the transaction is rolled back and thus no metadata changes are made.
 
+Inserting from a Query
+$$$$$$$$$$$$$$$$$$$$$$
+
+Applications like event data pipelines and real-time dashboards require fast queries across predefined aggregations of incoming data. One way to make these queries fast is by calculating and saving aggregates ahead of time. This is called "rolling up" the data and it avoids the cost of processing raw data at run-time. As an extra benefit, rolling up timeseries data into hourly or daily statistics can also save space. Old data may be deleted when its full details are no longer needed and aggregates suffice.
+
+For example, here is a table for tracking page views by url and a query to aggregate views per day.
+
+.. code-block:: postgresql
+
+  CREATE TABLE page_views (
+    site_id int,
+    url text,
+    host_ip inet,
+    view_time timestamp default now(),
+
+    PRIMARY KEY (site_id, url)
+  );
+
+  SELECT view_time::date AS day, site_id, url,
+         count(*) AS view_count
+  FROM page_views
+  GROUP BY view_time::date, site_id, url;
+
+One disadvantage of repeatedly executing the aggregate query is that it always has to recompute on the whole dataset. It's faster to save the information to a roll up :code:`daily_page_views` table and query that. We create the table and use :code:`INSERT INTO ... SELECT` to populate the table directly from the results of a query:
+
+.. code-block:: postgresql
+
+  CREATE TABLE daily_page_views (
+    site_id int,
+    day date,
+    url text,
+    view_count bigint,
+    PRIMARY KEY (site_id, day, url)
+  );
+
```

Comment:
(maybe then take a break from this code snippet and explain colocation a bit)

In this example, we distributed both :code:`page_views` and :code:`daily_page_views` on the `site_id` column. This ensures that data corresponding to a particular site will get co-located (link to colocation_groups) to the same node. Keeping the two tables' rows together on each node minimizes network traffic between nodes and enables highly parallel execution.

Once we created this new distributed table, we can then run INSERT INTO SELECT to roll up raw page views into the aggregated table. In the following, we aggregate page views each day. We find that users also wait for a certain time period after the end of day, to accommodate for late arriving data.